crates/task-impls/src/events.rs

```diff
@@ -62,4 +62,30 @@ pub enum SequencingHotShotEvent<TYPES: NodeType, I: NodeImplementation<TYPES>> {
     TransactionSend(TYPES::Transaction, TYPES::SignatureKey),
     /// Event to send DA block data from DA leader to next quorum leader (which should always be the same node); internal event only
     SendDABlockData(TYPES::BlockType),
+    /// Send VID shares to VID storage nodes; emitted by the DA leader
+    ///
+    /// Like [`DAProposalSend`].
+    VidDisperseSend(VidDisperse<TYPES>, TYPES::SignatureKey),
+    /// Vid disperse data has been received from the network; handled by the DA task
+    ///
+    /// Like [`DAProposalRecv`].
+    /// TODO GG: lots of boilerplate here eg. `VidDisperse[Send|Recv]`. If we need both [Send|Recv] then maybe partition [`SequencingHotShotEvent`] into separate enums?
+    /// TODO GG: should [`VidDisperse`] be of the form [`Proposal<DAProposal<TYPES>>`] to mirror [`DAProposalRecv`]? We need things like `Committable`, etc.
```

Comment:
It probably should, but also we need to revisit those traits - we are outgrowing them.  I ran into a similar issue with the timeout certificate logic where the exchange trait is too restrictive.  And the ViewSync logic works with those traits, but is a bit of a square peg in a round hole.

Reply:
new issue https://github.com/EspressoSystems/HotShot/issues/1694